time stamp
uint8[] data